src/guides/rpc_index.md

```diff
@@ -1,3 +1,80 @@
-# How-to Guides
+# RPC Overview
+
+An RPC node is a server that allows external applications to interact with the Polkadot chain. This interaction is facilitated through a set of standardized APIs which enable the execution of various operations and retrieval of blockchain data. Here are some key points about an RPC node in Polkadot:
+
+Functions of an RPC Node:
+*Querying Blockchain Data*: RPC nodes can be used to fetch information from the blockchain, such as block details, transaction histories, account balances, and other state information.
+*Submitting Transactions*: External applications can use RPC nodes to broadcast transactions to the network. This includes sending tokens, executing smart contracts, or any other transaction type supported by Polkadot.
+*Subscription to Events*: RPC nodes can provide real-time updates on blockchain events, such as new blocks being added, changes in account balances, or other relevant blockchain events.
+*Chain State and Metadata*: RPC nodes allow applications to query the current state of the blockchain, as well as metadata about the runtime environment, including the version of the Polkadot runtime, the list of available RPC methods, and more.
+
+### Type of RPC Nodes
+
+There are two common types of RPC nodes: *archive* nodes and *pruned* nodes, which differ in data storage and access.
+
+- **Archive Nodes**:
+  - Store the entire blockchain history, providing full state access for tasks requiring historical data, such as analytics and audits.
+  - Require substantial disk space.
+  - **Important Flag**: `--state-pruning archive --blocks-pruning archive` (Ensures the node retains all historical data)
```

Comment:
Just as a note, no need to add it to the docs as it will soon be enabled by default for archive nodes. 

There is also an offchain DB that you can store with the following flags:
```
--offchain-worker always
--enable-offchain-indexing true
```
Help:
```

      --offchain-worker <ENABLED>
          Execute offchain workers on every block
          
          [default: when-authority]

          Possible values:
          - always:         Always have offchain worker enabled
          - never:          Never enable the offchain worker
          - when-authority: Only enable the offchain worker when running as a validator (or
            collator, if this is a parachain node)

      --enable-offchain-indexing <ENABLE_OFFCHAIN_INDEXING>
          Enable offchain indexing API.
          
          Allows the runtime to write directly to offchain workers DB during block import.
          
          [default: false]
          [possible values: true, false]

```

Reply:
`--blocks-pruning archive` I don't this this is needed, only `--state-pruning archive`. @BulatSaif do you confirm ?

Reply:
> `--blocks-pruning archive` I don't this this is needed, only `--state-pruning archive`. @BulatSaif do you confirm ?

`--state-pruning` is for block's state while  `--blocks-pruning`  is for blocks body. Ideally in archive mode, you want to store all blocks not just finalized blocks. 
```
      --state-pruning <PRUNING_MODE>
          Specify the state pruning mode.

          This mode specifies when the block's state (ie, storage) should be pruned (ie,
          removed) from the database. This setting can only be set on the first creation
          of the database. Every subsequent run will load the pruning mode from the
          database and will error if the stored mode doesn't match this CLI value. It is
          fine to drop this CLI flag for subsequent runs. Possible values: - archive:
          Keep the state of all blocks. - 'archive-canonical' Keep only the state of
          finalized blocks. - number Keep the state of the last number of finalized
          blocks. [default: 256]

      --blocks-pruning <PRUNING_MODE>
          Specify the blocks pruning mode.

          This mode specifies when the block's body (including justifications) should be
          pruned (ie, removed) from the database. Possible values: - 'archive' Keep all
          blocks. - 'archive-canonical' Keep only finalized blocks. - number Keep the
          last `number` of finalized blocks.

          [default: archive-canonical]
```

Reply:
I never tested it, but `archive` and `archive-canonical` are very similar, so I think we can omit specifying it.